### DISCUSSION TITLE
change the value of variables

### DISCUSSION THREAD

**Discussion by Art-Che** [AutomaApp/automa#1035] [Q&A]
Hi!
I am getting a variable from the gettex block. Then, the if else code is executed.
After that, I need to create a new variable in the automaRefData function. And pass the text to the next block.
How can I do that?
![image](https://user-images.githubusercontent.com/120256667/209538485-a180e2c3-5e81-4056-a233-b26fbdb1b94c.png)

![image](https://user-images.githubusercontent.com/120256667/209538544-ae62a30c-c0c0-4ebe-971a-1bc6ce2c5f0b.png)

`var BIN = automaRefData('variables', 'BIN');
var BIN_length = (''+ BIN).length;

var i;
if (BIN_length < 12) {
    BIN_true = ("0" + BIN );}

else {BIN_true = BIN;
    }

automaRefData('variables', 'BIN_true')

automaNextBlock()`

**Comment by Kholid060** [comment]
Use the [`automaSetVariable`](https://docs.automa.site/blocks/javascript-code.html#automasetvariable-name-value) function

```js
automaSetVariable('BIN', BIN_true);
```

**Reply by Art-Che** [reply]
it worked, great! Thank you!
